### DISCUSSION TITLE
Select option custom attribute

### DISCUSSION THREAD

**Discussion by Sabyrzhanuly** [sa-si-dev/virtual-select#103] [Q&A] [answered]
How to add custom attribute to option and detect on change?

**Comment by sa-si-dev** [comment]
> How to add custom attribute to option and detect on change?

As of now, we could set only value and label properties in the options details.

I will support to add custom data with options details in the future release.

**Comment by sa-si-dev** [comment, marked as answer]
@Sabyrzhanuly you can set `customData` with options details and it would be returned on `getSelectedOptions()` method call.

```js
VirtualSelect.init({
  ele: '#sample-select',
  options: [
    { label: 'Options 1', value: '1', customData: 'text' },
    { label: 'Options 2', value: '2', customData: 6 },
    { label: 'Options 3', value: '3', customData: { cv1: true } },
  ],
});
```

This feature is available in the [release v1.0.26](https://github.com/sa-si-dev/virtual-select/releases/tag/v1.0.26)

**Comment by pjasminejeni** [comment]
Hello, If we need to add customData directly to the options instead of initiating from jquery, What I should do? Say,
                        ![image](https://user-images.githubusercontent.com/112079018/232453205-9358871a-123a-4b8c-95bd-a7967ea0d306.png)

Thanks in advance :-)

**Comment by sa-si-dev** [comment]
> Hello, If we need to add customData directly to the options instead of initiating from jquery, What I should do? Say,
>                         ![image](https://user-images.githubusercontent.com/112079018/232453205-9358871a-123a-4b8c-95bd-a7967ea0d306.png)
> 
> Thanks in advance :-)

@pjasminejeni you can use [options[].customData](https://sa-si-dev.github.io/virtual-select/#/properties?id=properties)

**Reply by pjasminejeni** [reply]
Thank you @sa-si-dev :-)
It will be better if I get an example :-D
